umbraco/Umbraco.CMS.Backoffice
# Block Grid: Correct settings data if config changed

This PR ensures that the data of a Block Editor corresponds to the configuration, Well, only when revisited in the UI.

This also corrects document type picker + its property editor, so its value is undefined when empty. before it gave an empty string.


## Types of changes




## Motivation and context



## How to test?

* Make a docType with a Block Editor
* setup a Block Type with a settings model
* Make a document of this type and create the data.
* Remove the settings model from the Block Type.
* Revisit the page to see that UI removes the data. you could save it, to inspect the Network calls, compare before with after.
* There might be other similar cases that needs to be tested as well.

## Screenshots (if appropriate)

## Checklist
